internal/deprecation/deprecation.go

```diff
@@ -131,7 +132,7 @@ func (checker *Checker) check(versionNumber string) (*Info, *failures.Failure) {
 
 func (checker *Checker) shouldFetch() bool {
 	lastFetch := checker.config.GetTime(fetchKey)
-	if !lastFetch.IsZero() && time.Now().Before(lastFetch) {
+	if !lastFetch.IsZero() && time.Now().Before(lastFetch) && fileutils.FileExists(checker.deprecationFile) {
```

Comment:
This feels like the wrong place for this check. This conditional is about asserting whether it's time to do a check. Whether this file exists should not be part of this assertion. I think you need this check a little later in the process.

Reply:
Fair enough. I moved the check for the file to before this timed check as it makes sense to fetch the info if the file doesn't exist regardless of the time of the last fetch.